### PR TITLE
GraphQL schema setup with User and DailyStat

### PR DESCRIPTION
**Pull request by achhetr**
UserType and DailyStatType added to the graphql schema

getDailyStatById method added to graphql to fetch stats by id

### CODE REVIEW

File: app/controllers/graphql_controller.rb [hunk 1]
```diff
@@ -9,8 +9,7 @@ def execute
     query = params[:query]
     operation_name = params[:operationName]
     context = {
-      # Query context goes here, for example:
-      # current_user: current_user,
+      current_user: current_user,
```
Comment (saramic) [inline app/controllers/graphql_controller.rb:12]: do we also need to force authorisation? what happens when you run graphql when you are NOT logged in? should we look at authenticating via JWT? as per example something like https://github.com/saramic/real-code-runner/commit/9af6752aa4d15ff59a9153cb24bdf91adb49a71b

File: app/graphql/types/query_type.rb [hunk 2]
```diff
@@ -1,17 +1,31 @@
...
-                               description: "An example field added by the generator"
-    def test_field
-      "Hello World!"
+    field :daily_stat_by_id, DailyStatType, null: true do
+      description "get daily stats of the user by daily stats id"
+      argument :id, ID, required: true
+    end
+
+    def daily_stat_by_id(id:)
+      DailyStat.find(id)
```
Comment (saramic) [inline app/graphql/types/query_type.rb:18]: how do we plan to protect this to the signed in user? should it be

```
context[:current_user].daily_stats.find_by(id: id)
```

or

```
DailyStat.find_by(id: id, user_id: context[:current_user].id)
```

?

File: app/graphql/types/user_type.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    # field :last_sign_in_ip, String, null: true
+    # field :confirmation_token, String, null: true
+    # field :confirmed_at, GraphQL::Types::ISO8601DateTime, null: true
+    # field :confirmation_sent_at, GraphQL::Types::ISO8601DateTime, null: true
+    # field :unconfirmed_email, String, null: true
+    # field :failed_attempts, Integer, null: false
+    # field :unlock_token, String, null: true
+    # field :locked_at, GraphQL::Types::ISO8601DateTime, null: true
+    # field :created_at, GraphQL::Types::ISO8601DateTime, null: false
+    # field :updated_at, GraphQL::Types::ISO8601DateTime, null: false
```
Comment (saramic) [inline app/graphql/types/user_type.rb:22]: YAGNI - delete it - we can add it when we actually need it

File: app/javascript/api/ApolloClient.js [hunk 4]
```diff
@@ -1,25 +1,21 @@
-import {
-	ApolloClient,
-	createHttpLink,
-	InMemoryCache,
-} from '@apollo/client';
+import { ApolloClient, createHttpLink, InMemoryCache } from '@apollo/client';
```
Comment (saramic) [inline app/javascript/api/ApolloClient.js:1]: why are we getting inconsistent linting??? I am not sure

File: app/javascript/react/components/App.js [hunk 5]
```diff
@@ -6,8 +6,8 @@ import ApiDemo from './ApiDemo/ApiDemo';
 
 const App = () => (
   <ApolloProvider client={ApolloClient}>
-    <LayoutComponent />
-    <ApiDemo />
-  </ApolloProvider>
+  <LayoutComponent />
+  <ApiDemo />
+	</ApolloProvider>
```
Comment (saramic) [inline app/javascript/react/components/App.js:11]: bad formatting - and is this tabs again????

File: app/javascript/react/components/LayoutComponent/LayoutComponent.js [hunk 6]
```diff
@@ -2,13 +2,11 @@ import React, { Fragment } from 'react';
...
-      <Tabs />
-    </Fragment>
-  );
-};
+const LayoutComponent = () => (
+	<Fragment>
+		<Header />
+		<Tabs />
+  </Fragment>
+);
```
Comment (saramic) [inline app/javascript/react/components/LayoutComponent/LayoutComponent.js:10]: more tabs?

File: app/javascript/react/components/MainComponent/MainComponent.js [hunk 7]
```diff
@@ -2,13 +2,11 @@ import React, { Fragment } from 'react';
...
-      <Tabs />
-    </Fragment>
-  );
-};
+const MainComponent = () => (
+	<Fragment>
+		<Header />
+		<Tabs />
+  </Fragment>
+);
```
Comment (saramic) [inline app/javascript/react/components/MainComponent/MainComponent.js:10]: tabs??

**Review by saramic — COMMENTED** [review]
let's try to sort out our tabs, and there are some questions about how we handle finding stats for a user

File: app/javascript/api/ApolloClient.js [hunk 8]
```diff
@@ -1,25 +1,21 @@
...
 const httpLink = createHttpLink({
 	uri: '/graphql',
 });
 
 const authLink = setContext((_, { headers }) => ({
 	headers: {
-		// 'X-CSRF-Token': xCsrfToken(),
-		// AUTHORIZATION: `Token ${process.env.GRAPHQL_API_KEY}`,
+		'X-CSRF-Token': xCsrfToken(),
+		AUTHORIZATION: `Token ${process.env.GRAPHQL_API_KEY}`,
```
[the remark this answers is not in the document]
Comment (achhetr) [reply app/javascript/api/ApolloClient.js:18]: without it, it was giving csrf authentication error

File: app/graphql/types/query_type.rb (hunk 2 above)
Comment (achhetr) [reply app/graphql/types/query_type.rb:18]: great question

**Pull request merged**